BDArmory/Control/BDModulePilotAI.cs

```diff
@@ -2301,7 +2301,14 @@ void FlyToTargetVessel(FlightCtrlState s, Vessel v)
                         if (distanceToTarget > Mathf.Max(4500f, extendDistanceAirToGround + ((float)vessel.horizontalSrfSpeed * BDAMath.Sqrt(2 * finalBombingAlt / bodyGravity)) + finalBombingAlt)) //lead based on estimate of fall time at desired alt, regardless if we're there yet
                         {
                             finalMaxSteer = GetSteerLimiterForSpeedAndPower();
-                            target = target + (finalBombingAlt * upDirection); //aim for target alt while still out of range
+                            //target = target + (finalBombingAlt * upDirection); //aim for target alt while still out of range
+                            if (missile.GetWeaponClass() != WeaponClasses.SLW) //semi-aggressively get to desired bombing alt before we get into range
+                            {
+                                if (Mathf.Abs((float)vessel.altitude - finalBombingAlt) > 100) target = transform.position + (target - transform.position).normalized * 2000; //get to bombing alt if not yet there.
```

Comment:
I was thinking more along the lines of something like this:
```csharp
target = AIUtils.PredictPosition(v, weaponManager.bombAirTime); //make AI properly lead bombs vs moving targets, also why AI didn't like dropping them before. Should be at altitude, so use correct timing
var (distance, direction) = (vessel.CoM - target).ProjectOnPlanePreNormalized(upDirection).MagNorm();
target = target + 0.5f * distance * direction + finalBombingAlt * upDirection; // Aim for the bombing altitude at half-way to the target
```
which might work well for torp bombing too, though maybe with a larger factor like 0.75?

(I was away from the PC when you updated the PR.)

Reply:
Possibly. Having both use the same equation, just different factor would simplify the code.

Reply:
Seems to work fine for both level and torp bombing alt changes